packages/dds/map/src/interfaces.ts

```diff
@@ -191,18 +190,36 @@ export interface IDirectory extends Map<string, any> {
     getWorkingDirectory(relativePath: string): IDirectory;
 }
 
+/**
+ * Interface describing actions on a directory.
+ *
+ * @remarks
+ * When used as a Map, operates on its keys.
+ */
+export interface IDirectory extends IDirectoryNoEvents, IEventProvider<IDirectoryEvents> { }
+
 export interface ISharedDirectoryEvents extends ISharedObjectEvents {
     (event: "valueChanged", listener: (
         changed: IDirectoryValueChanged,
         local: boolean,
         op: ISequencedDocumentMessage,
-        target: IEventThisPlaceHolder) => void);
+        target: IEventThisPlaceHolder,
+    ) => void);
+}
+
+export interface IDirectoryEvents extends IEvent {
+    (event: "containedValueChanged", listener: (
+        changed: IValueChanged,
+        local: boolean,
+        op: ISequencedDocumentMessage,
+        target: IEventThisPlaceHolder,
```

Comment:
can we stop progagating the op. i think it just creates more problems as people think they can just use the values as is, but thats not really the case, and not sure what the uses case is

Reply:
Good point

Reply:
Removed in latest